pkg/cmd/context/generate/build-configs.go

```diff
@@ -0,0 +1,101 @@
+package generate
+
+import (
+	"github.com/redhat-developer/app-services-cli/pkg/core/ioutil/icon"
+	"github.com/redhat-developer/app-services-cli/pkg/core/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/core/servicecontext"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/contextutil"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+
+	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+)
+
+type configValues struct {
+	KafkaURL     string
+	RegistryURL  string
+	ClientID     string
+	ClientSecret string
+}
+
+func createServiceAccount(opts *options, shortDescription string) (*kafkamgmtclient.ServiceAccount, error) {
+
+	conn, err := opts.Connection(connection.DefaultConfigSkipMasAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	serviceAccountPayload := kafkamgmtclient.ServiceAccountRequest{Name: shortDescription}
+
+	serviceacct, httpRes, err := conn.API().
+		ServiceAccountMgmt().
+		CreateServiceAccount(opts.Context).
+		ServiceAccountRequest(serviceAccountPayload).
+		Execute()
+
+	if httpRes != nil {
+		defer httpRes.Body.Close()
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &serviceacct, nil
+}
+
+// BuildConfiguration builds the configs for the service context
+func BuildConfiguration(svcConfig *servicecontext.ServiceConfig, opts *options) error {
+
+	factory := &factory.Factory{
+		IOStreams:      opts.IO,
+		Logger:         opts.Logger,
+		Context:        opts.Context,
+		Localizer:      opts.localizer,
+		Connection:     opts.Connection,
+		ServiceContext: opts.ServiceContext,
+	}
+
+	configurations := &configValues{}
+
+	var serviceAvailable bool
+
+	kafkaInstance, _ := contextutil.GetCurrentKafkaInstance(factory)
```

Comment:
not handled error

Reply:
What if use wants configuration to be generated with only one service?

Reply:
Generating config on invalid context should not be allowed. Can we unset service from context?

Reply:
Like this would 100% be logged as bug. I have my service in context but not getting url in config. See my handling for status that does similar thing. One service breaking context returns error for entire process no matter what

Reply:
#1498

Reply:
Unsetting a service is not there yet. A freshly created context can have an unset service.

Reply:
Actually I see problem now. When registry is not set it will return error
but for this case this is ok.

Reply:
I would say that usage of those helper methods doesn't fit to this use case. We actually need to duplicate logic for those for purpose of generation.